### PR TITLE
Fix/stad 107 update interval

### PR DESCRIPTION
**Pull request by muthenberg**

### CODE REVIEW

File: DataCapturing.podspec [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 Pod::Spec.new do |s|
   s.name             = 'DataCapturing'
-  s.version          = '4.2.3'
+  s.version          = '4.3.0'
```
Comment (muthenberg) [inline DataCapturing.podspec]: new function to set the update interval for the event handler justifies increase in second place.

File: DataCapturing/DataCapturingService.swift [hunk 2]
```diff
@@ -26,7 +26,7 @@ import os.log
  An object of this class handles the lifecycle of starting and stopping data capturing.
  
  - Author: Klemens Muthmann
- - Version: 9.1.0
+ - Version: 9.2.0
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: Same argumentation as for the complete version here.

File: DataCapturing/DataCapturingService.swift [hunk 3]
```diff
@@ -41,14 +41,11 @@ public class DataCapturingService: NSObject {
...
     public var isPaused: Bool
 
-    /// A listener that is notified of important events during data capturing.
-    private var handler: ((DataCapturingEvent, Status) -> Void)
-
     /// The currently recorded `Measurement` or `nil` if there is no active recording.
     public var currentMeasurement: MeasurementEntity?
 
-    /// An instance of `CMMotionManager`. There should be only one instance of this type in your application.
-    private let motionManager: CMMotionManager
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: this has moved to the other private attributes

File: DataCapturing/DataCapturingService.swift [hunk 4]
```diff
@@ -41,14 +41,11 @@ public class DataCapturingService: NSObject {
...
+    /// Locations are captured approximately once per second on most devices. If you would like to get fewer updates this parameter controls, how many events are skipped before one is reported to your handler. The default value is 1, which reports every event. To receive fewer events you could for example set it to 5 to only receive every fifth event.
+    public var locationUpdateSkipRate = 1
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: this is the actual important update within this commit.

File: DataCapturing/DataCapturingService.swift [hunk 5]
```diff
@@ -41,14 +41,11 @@ public class DataCapturingService: NSObject {
     /// `true` if data capturing was running but is currently paused; `false` otherwise.
     public var isPaused: Bool
 
-    /// A listener that is notified of important events during data capturing.
-    private var handler: ((DataCapturingEvent, Status) -> Void)
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: grouped with other private attributes

File: DataCapturing/DataCapturingService.swift [hunk 6]
```diff
@@ -92,6 +95,9 @@ public class DataCapturingService: NSObject {
     /// A timer called in regular intervals to save the captured data to the underlying database.
     private var backgroundSynchronizationTimer: DispatchSourceTimer!
 
+    /// The number of the current event. This is used to filter events based on `locationUpdateRate`.
+    private var geoLocationEventNumber = 0
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: this is a counter for how many geo location update events have already been skipped. It is set to 0 when it reaches locationUpdateSkipRate

File: DataCapturing/DataCapturingService.swift [hunk 7]
```diff
@@ -161,8 +167,8 @@ public class DataCapturingService: NSObject {
      running.
 
      - Throws:
-        - `DataCapturingError.isPaused` if the service was paused and thus stopping it makes no sense.
-        - `DataCapturingError.noCurrentMeasurement` If there is no measurement to stop.
+        - `PersistenceError` If the currently captured measurement was not found in the database.
+        - Some unspecified errors from within *CoreData*.
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: documentation was wrong here. These are the true exceptions thrown at this place.

File: DataCapturing/DataCapturingService.swift [hunk 8]
```diff
@@ -216,9 +222,9 @@ public class DataCapturingService: NSObject {
      Resumes the current data capturing with the data capturing measurement that was running when `pause()` was called. A call to this method is only valid after a call to `pause()`. It is going to fail if used after `start()` or `stop()`.
 
      - Throws:
-     - `DataCapturingError.notPaused`: If the service was not paused and thus resuming it makes no sense.
-     - `DataCapturingError.isRunning`: If the service was running and thus resuming it makes no sense.
-     - `DataCapturingError.noCurrentMeasurement`: If no current measurement is available while resuming data capturing.
+        - `DataCapturingError.notPaused`: If the service was not paused and thus resuming it makes no sense.
+        - `DataCapturingError.isRunning`: If the service was running and thus resuming it makes no sense.
+        - `DataCapturingError.noCurrentMeasurement`: If no current measurement is available while resuming data capturing.
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: stupid Xcode auto formatter always removes these. But they need to be there for correct documentation formatting.

File: DataCapturing/DataCapturingService.swift [hunk 9]
```diff
@@ -368,31 +374,46 @@ extension DataCapturingService: CLLocationManagerDelegate {
...
+        - manager: The location manager used.
+        - didUpdateLocation: An array of the updated locations.
      */
     public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
 
         for location in locations {
             // Smooth the way by removing outlier coordinates.
             let howRecent = location.timestamp.timeIntervalSinceNow
             guard location.horizontalAccuracy < 20 && abs(howRecent) < 10 else { continue }
+            let timestamp = DataCapturingService.convertToUtcTimestamp(date: location.timestamp)
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: moved this out for easier debugging.

File: DataCapturing/DataCapturingService.swift [hunk 10]
```diff
@@ -368,31 +374,46 @@ extension DataCapturingService: CLLocationManagerDelegate {
...
+                        return
+                    }
+
+                    self.handler(.geoLocationAcquired(position: geoLocation), .success)
+                }
             }
+            if geoLocationEventNumber == locationUpdateSkipRate {
+                geoLocationEventNumber = 0
+            }
+
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: this part was the core of the update.

File: DataCapturingTests/DataCapturingTests.swift [hunk 11]
```diff
@@ -26,7 +26,7 @@ import CoreData
  This test is intended to test capturing some data in isolation.
```
Comment (muthenberg) [inline DataCapturingTests/DataCapturingTests.swift]: And finally some test. :D Working Tests make me happy!!!

File: docs/Classes.html [hunk 12]
```diff
@@ -13,7 +13,7 @@
     <a title="Classes  Reference"></a>
```
Comment (muthenberg) [inline docs/Classes.html]: From here on there is only generated documentation, which just mirrors the inline documentation. Just ignore it.

File: DataCapturing/DataCapturingService.swift (hunk 10 above)
Comment (hb0) [reply DataCapturing/DataCapturingService.swift]: Maybe throw an IllegalArgumentException if someone defines a locationUpdateRate < 1;

**Pull request merged**